### ISSUE TITLE
Trouble on Mobile Safari

### ISSUE THREAD

**Issue by micky2be** [stacktracejs/error-stack-parser#16]
Running a simple `ErrorStackParser.parse(new Error('boom'));` on iOS9 and iOS8.3 generate an error >> 'Cannot parse given Error object'

**Comment by oliversalzburg** [comment]
Can you provide a sample of `new Error('boom').stack`? I don't have an iOS device handy :(

**Comment by micky2be** [comment]
```bash
"eval code
eval@[native code]
_evaluateOn
_evaluateAndWrap
evaluate"
```

**Comment by oliversalzburg** [comment]
That's really weird. What code is generating this? What output do you get from http://putr.de/~oliver/error?

**Comment by micky2be** [comment]
That's from the console

**Comment by eriwen** [comment]
Hm... that's no good. We may be able to switch to iOS9 on Sauce for testing. I'll look into this one.

**Comment by oliversalzburg** [comment]
I get pretty well formatted stack traces on iOS 9. Tested it earlier with http://putr.de/~oliver/error on one of our dev devices.

**Comment by eriwen** [comment]
I think what @micky2be is saying is that typing `ErrorStackParser.parse(new Error('boom'));` directly into an iOS JS console causes this error. 

Realistically, there's no useful information in the `stack`, but maybe error-stack-parser can return with one stack frame for `eval` instead of returning an Error

**Comment by oliversalzburg** [comment]
@eriwen Yeah, my assumption was that the regular expressions simply don't match the stack trace. But I don't understand why that is, because the produced stack traces look fine.

Is this specific to code being run directly from the console? If so, I might be able to test it tomorrow, when I find out how to access the console ;D

**Comment by micky2be** [comment]
I have same trouble with my running code. 
I had to put a try catch around to avoid troubles.

**Comment by oliversalzburg** [comment]
I put up a minimal setup at http://putr.de/~oliver/esp/ and it works as intended on iOS9.0.2 and 7.0.2. A code sample that produces the issue would be really helpful.

**Comment by micky2be** [comment]
Ok, so I think I put you guys in the wrong direction.
On Mobile Safari I do get from http://putr.de/~oliver/error:
```
throwError@http://putr.de/~oliver/error:9:17
global code@http://putr.de/~oliver/error:4:12
```

But when using a built code (commonJS) with sourceURL I only get:
```
throwError
```

So not really a problem from error-stack-parser then :/

**Comment by eriwen** [comment]
@micky2be Are you using stacktrace.js or just error-stack-parser?

**Comment by micky2be** [comment]
error-stack-parser

Can't really work with stacktrace.js at the moment because of promises and component.json being malformed. 
Which remind me I have to a PR for that.

**Comment by oliversalzburg** [comment]
@micky2be Please also tell us about any issues with the promises and `component.json`.

**Comment by eriwen** [comment]
This recent commit (https://github.com/stacktracejs/error-stack-parser/commit/53dd6f8877ac1c3c335257f50a57179ddd47376b) included in v1.3.1 should actually fix this issue, which seems to have been happening because newer Safari versions sometimes don't include location information (within `eval`, for example).

@micky2be would you be willing to confirm this?

**Comment by micky2be** [comment]
Sadly it's not fixing the issue. The problem is right before that:
https://github.com/stacktracejs/error-stack-parser/blob/1.3.1/dist/error-stack-parser.js#L26-L35
The stack is not detected as anything.

Beside the regex I see in the commit will detect stack only if the error is executed from the console.

When adding a `console.error('Hello')` in my project I get the following stack:
```
_createLanguageMenuButton
_updateContent
undefined
g
emit
show
backToLogin
undefined
undefined
undefined
undefined
_arrayEach
taskComplete
onNextTick
run
drainQueue
```

Maybe you could add the following at the line 33
```js
} else if (error.stack) {
    return this.parseFFOrSafari(error);
}
```
Then yes, in that case I would get something.
